### PR TITLE
Bluetooth: Testlib: ATT read fixes

### PR DESCRIPTION
**Pull request by alwa-nordic**
See individual commits

### CODE REVIEW

**Review by jori-nordic — APPROVED** [review]

**Review by theob-pro — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

File: tests/bluetooth/common/testlib/src/att_read.c [hunk 1]
```diff
@@ -167,17 +174,16 @@ int bt_testlib_gatt_long_read(struct net_buf_simple *result_data, uint16_t *resu
...
-		.conn = conn,
-		.result_data = result_data,
-		.params = {
-			.handle_count = 1,
-			.single = {.handle = handle, .offset = offset},
-			IF_ENABLED(CONFIG_BT_EATT, (.chan_opt = bearer,))
-		}};
+	struct bt_testlib_att_read_closure ctx = {};
+
+	ctx.att_mtu = result_att_mtu;
```
Comment (LingaoM) [inline tests/bluetooth/common/testlib/src/att_read.c:179]: It broke bbsim CI https://github.com/zephyrproject-rtos/zephyr/actions/runs/7099469135/job/19323552333?pr=66136#step:12:997

Comment (jori-nordic) [reply tests/bluetooth/common/testlib/src/att_read.c:179]: fixing it in https://github.com/zephyrproject-rtos/zephyr/pull/66012
feel free to cherry-pick

Comment (LingaoM) [reply tests/bluetooth/common/testlib/src/att_read.c:179]: @alwa-nordic Ping.

Comment (LingaoM) [reply tests/bluetooth/common/testlib/src/att_read.c:179]: OK, thanks.

**Pull request merged**